src/Queue/AsyncAwsSqsQueue.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Laravel\Vapor\Queue;
+
+use AsyncAws\Sqs\SqsClient;
+use Illuminate\Contracts\Queue\Queue as QueueContract;
+use Illuminate\Queue\Jobs\SqsJob;
+use Illuminate\Queue\Queue;
+use Laravel\Vapor\Queue\Job\AsyncAwsSqsJob;
+
+class AsyncAwsSqsQueue extends Queue implements QueueContract
+{
+    /**
+     * The Amazon SQS instance.
+     *
+     * @var \AsyncAws\Sqs\SqsClient
+     */
+    protected $sqs;
+
+    /**
+     * The name of the default queue.
+     *
+     * @var string
+     */
+    protected $default;
+
+    /**
+     * The queue URL prefix.
+     *
+     * @var string
+     */
+    protected $prefix;
+
+    /**
+     * Create a new Amazon SQS queue instance.
+     *
+     * @param  \AsyncAws\Sqs\SqsClient  $sqs
+     * @param  string  $default
+     * @param  string  $prefix
+     * @return void
+     */
+    public function __construct(SqsClient $sqs, $default, $prefix = '')
+    {
+        $this->sqs = $sqs;
+        $this->prefix = $prefix;
+        $this->default = $default;
+    }
+
+    /**
+     * Get the size of the queue.
+     *
+     * @param  string|null  $queue
+     * @return int
+     */
+    public function size($queue = null)
+    {
+        $response = $this->sqs->getQueueAttributes([
+            'QueueUrl' => $this->getQueue($queue),
+            'AttributeNames' => ['ApproximateNumberOfMessages'],
```

Comment:
Could use the constant instead `QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES`

Reply:
Sure. Thank you